context/FormProvider.tsx

```diff
@@ -0,0 +1,158 @@
+import { useAccount, useContractRead } from "@starknet-react/core";
+import React, { FunctionComponent, useState } from "react";
+import { createContext, useMemo } from "react";
+import { computeMetadataHash, generateSalt } from "@/utils/userDataService";
+import { useNftPpVerifierContract } from "@/hooks/contracts";
+import { Abi } from "starknet";
+import { filterAssets, retrieveAssets } from "@/utils/nftService";
+
+interface FormState {
+  email: string;
+  isSwissResident: boolean;
+  tokenId: number;
+  duration: number;
+  selectedDomains: Record<string, boolean>;
+  // metadata
+  salt?: string;
+  metadataHash?: string;
+  needMetadata: boolean;
+  // tax
+  salesTaxRate: number;
+  //pfp
+  selectedPfp?: StarkscanNftProps;
+  //upsell
+  isUpselled: boolean;
+}
```

Comment:
Super cool! 

Why an interface and not a type ?

Reply:
I think both work for context